src/content/datachannel/messaging/main.js

```diff
@@ -0,0 +1,164 @@
+/*
+ *  Copyright (c) 2018 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+/* eslint no-unused-expressions: 0 */
+
+'use strict';
+import {LitElement, html} from 'https://unpkg.com/@polymer/lit-element?module';
+
+class MessagingSample extends LitElement {
+  constructor() {
+    super();
+    this._localConnection = this._remoteConnection = this._localChannel = this._remoteChannel = null;
+    this.connected = false;
+  }
+
+  disconnect() {
+    this._localConnection.close();
+    this._remoteConnection.close();
+  }
+
+  async connect() {
+    try {
+      const dataChannelParams = {ordered: true};
+      this._localConnection = new RTCPeerConnection();
+      this._localConnection.addEventListener('icecandidate', async e => {
+        console.log('local connection ICE candidate: ', e.candidate);
+        this._remoteConnection.addIceCandidate(e.candidate);
+      });
+      this._remoteConnection = new RTCPeerConnection();
+      this._remoteConnection.addEventListener('icecandidate', async e => {
+        console.log('remote connection ICE candidate: ', e.candidate);
+        this._localConnection.addIceCandidate(e.candidate);
+      });
+
+      this._localChannel = this._localConnection
+        .createDataChannel('messaging-channel', dataChannelParams);
+      this._localChannel.binaryType = 'arraybuffer';
+      this._localChannel.addEventListener('open', () => {
+        console.log('Local channel open!');
+        this.connected = true;
+      });
+      this._localChannel.addEventListener('close', () => {
+        console.log('Local channel closed!');
+        this.connected = false;
+      });
+      this._localChannel.addEventListener('message', e => this._onLocalMessageReceived(e));
+
+      this._remoteConnection.addEventListener('datachannel', e => this._onRemoteDataChannel(e));
+
+      const localOffer = await this._localConnection.createOffer();
```

Comment:
Could you do the remoteOffer in parallel instead of waiting? ie 

```
const initLocalOffer = async () => {
   ...
}

const initRemoteOffer = async () => {
   ...
}

await Promise.all([initLocalOffer, initRemoteOffer);
console.log("All initialized!")
```

Reply:
Tested this and that doesn't work. The result from `createOffer()` must be set before we can call `createAnswer()`.

Reply:
Then do

```
const initLocalOffer = async () => {
   ...
}

const initRemoteOffer = async () => {
   ...
}

await initLocalOffer();
await initRemoveOffer();
```

At least that makes it clear what code belongs where